CMake: No target "gie"

When configuring a CMake build of PROJ with:
```
cmake -DBUILD_GIE=OFF -DBUILD_TESTING=ON ..
```
an error is repeated over 30 times:
```
CMake Error at cmake/ProjTest.cmake:31 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE_NAME:gie>

  No target "gie"
Call Stack (most recent call first):
  test/CMakeLists.txt:6 (proj_add_gie_test)
```
Some options are:

1. Skip tests that require gie
2. Build gie so it can be used for tests, but don't install it

Unless anyone has an objection, I'll go for option 2.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -70,7 +70,7 @@ if(BUILD_PROJINFO)
 endif()
 
 # Always build gie if testing is requested
-if(BUILD_TESTING)
+if(BUILD_GIE OR BUILD_TESTING)
   include(bin_gie.cmake)
   set(BIN_TARGETS ${BIN_TARGETS} gie)
 endif()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,7 +69,8 @@ if(BUILD_PROJINFO)
   set(BIN_TARGETS ${BIN_TARGETS} binprojinfo)
 endif()
 
-if(BUILD_GIE)
+# Always build gie if testing is requested
+if(BUILD_TESTING)
   include(bin_gie.cmake)
   set(BIN_TARGETS ${BIN_TARGETS} gie)
 endif()

--- a/src/bin_gie.cmake
+++ b/src/bin_gie.cmake
@@ -11,8 +11,10 @@ add_executable(gie ${GIE_SRC} ${GIE_INCLUDE})
 target_link_libraries(gie PRIVATE ${PROJ_LIBRARIES})
 target_compile_options(gie PRIVATE ${PROJ_CXX_WARN_FLAGS})
 
-install(TARGETS gie
-  DESTINATION ${BINDIR})
+if(BUILD_GIE)
+  install(TARGETS gie
+    DESTINATION ${BINDIR})
+endif()
 
 if(MSVC AND BUILD_SHARED_LIBS)
   target_compile_definitions(gie PRIVATE PROJ_MSVC_DLL_IMPORT=1)

